src/Core/Domain/Cart/QueryHandler/GetCartSummaryForOrderCreationHandler.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Cart\QueryHandler;
+
+use PrestaShop\PrestaShop\Core\Domain\Cart\Command\CreateEmptyCustomerCartCommand;
+use PrestaShop\PrestaShop\Core\Domain\Cart\CommandHandler\CreateEmptyCustomerCartHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Cart\Query\GetCartSummary;
+use PrestaShop\PrestaShop\Core\Domain\Cart\Query\GetCartSummaryForOrderCreation;
+use PrestaShop\PrestaShop\Core\Domain\Cart\Query\GetLastEmptyCustomerCart;
+
+final class GetCartSummaryForOrderCreationHandler implements GetCartSummaryForOrderCreationHandlerInterface
+{
+    /**
+     * @var GetLastEmptyCustomerCartHandlerInterface
+     */
+    private $getLastEmptyCustomerCartHandler;
+
+    /**
+     * @var CreateEmptyCustomerCartHandlerInterface
+     */
+    private $createEmptyCartForCustomerHandler;
+
+    /**
+     * @var GetCartSummaryHandlerInterface
+     */
+    private $getCartSummaryHandler;
+
+    public function __construct(
+        GetLastEmptyCustomerCartHandlerInterface $getLastEmptyCustomerCartHandler,
+        CreateEmptyCustomerCartHandlerInterface $createEmptyCartForCustomerHandler,
+        GetCartSummaryHandlerInterface $getCartSummaryHandler
+    ) {
+        $this->getLastEmptyCustomerCartHandler = $getLastEmptyCustomerCartHandler;
+        $this->createEmptyCartForCustomerHandler = $createEmptyCartForCustomerHandler;
+        $this->getCartSummaryHandler = $getCartSummaryHandler;
+    }
+
+    public function handle(GetCartSummaryForOrderCreation $query)
+    {
+        $customerId = $query->getCustomerId();
+
+        $cartId = $this->getLastEmptyCustomerCartHandler->handle(
+            new GetLastEmptyCustomerCart($customerId->getValue())
+        );
+
+        if (null === $cartId) {
+            $cartId = $this->createEmptyCartForCustomerHandler->handle(
+                new CreateEmptyCustomerCartCommand($customerId->getValue())
```

Comment:
the more we add commands, the more it seems that suffixing them with `Command` is redundant, i'd like to get rid of them, wdyt?

Reply:
1) You're right, you are violating the principle but you do this in order to follow legacy behavior so that's OK. In a 2nd step refactoring we could really split the 2 actions.

2) The Command suffix will be very useful for agencies and modules developers to find what they can use (I plan to generate a list of all available Commands in the doc using the name actually) 😉 so if it's not too bothering ... I'dl like to keep it.